Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM eclipse-temurin:21-jdk-alpine
+
+WORKDIR /app
+
+COPY .mvn/ .mvn
+COPY mvnw pom.xml ./
+RUN ./mvnw dependency:resolve
+
+COPY src ./src
+
+CMD ["./mvnw", "spring-boot:run"]
```

Comment:
**Suggestion:** Consider using multi-stage builds to reduce the size of the final Docker image. This can be done by separating the build stage and the application stage into two different stages. In the build stage, you can download all the necessary dependencies and build the application. In the application stage, you can copy the built application from the build stage and run it. This way, the final Docker image will not contain any unnecessary files and will be much smaller in size. [enhancement]
```suggestion
# Build stage
FROM eclipse-temurin:21-jdk-alpine AS build
WORKDIR /app
COPY .mvn/ .mvn
COPY mvnw pom.xml ./
RUN ./mvnw dependency:resolve
COPY src ./src
RUN ./mvnw package

# Application stage
FROM eclipse-temurin:21-jdk-alpine
WORKDIR /app
COPY --from=build /app/target/*.jar ./app.jar
CMD ["java", "-jar", "./app.jar"]
```